client/src/languagePlugins/javascript/javascriptPlugin.ts

```diff
@@ -33,23 +34,23 @@ class JavascriptBlockKind implements Langs.Block {
         ts.ScriptTarget.Latest
       );
       let tree = [];
+      
       for (var n=0; n < tsSourceFile.statements.length; n++){
         let node = tsSourceFile.statements[n];
-        // console.log(node)
         switch (node.kind) {
-          case 212: {
-              tree.push({
-                kind: 212,
-                name: node.declarationList.declarations[0].name,
-                initializer: node.declarationList.declarations[0].initializer
-              });
-            break
-          }
-          case 214: {
+          case ts.SyntaxKind.VariableStatement: {
             tree.push({
-              kind: 214,
-              expression: node.expression});
-            break;
+              kind: ts.SyntaxKind.VariableStatement,
+              name: node.declarationList.declarations[0].name,
+              initializer: node.declarationList.declarations[0].initializer
+            });
+          break
+          }
+        case 214: {
```

Comment:
I don't think we need any special handling for the expression statement case - I believe we should just run the `walk` function on all statements that we get back from `createSourceFile`.

There are many other kinds of statements like loops. I suspect something like the following will break the current version:

```
var a = 1;
```

```
var c = 0;
for(var i = 0; i<10; i++) { c = c+a; }
```

I think we will not correctly detect `a` as antecedent here.

Reply:
I think we need a case for ts.SyntaxKind.VariableStatement `(var a = 1;)` because it has the variable name, which is not found in any other ast node types. 
```
for (var n=0; n < tsSourceFile.statements.length; n++){
        let node = tsSourceFile.statements[n];
        switch (node.kind) {
          case ts.SyntaxKind.VariableStatement: {
            tree.push({
              kind: ts.SyntaxKind.VariableStatement,
              name: node.declarationList.declarations[0].name,
              initializer: node.declarationList.declarations[0].initializer
            });
          break
          }
          default: {
            tree.push(node);
          break;
          }
        }
      }
```
Then we can use it for the exportNode like this:
```
for (var s = 0; s < tree.length; s++) {
        let statement = tree[s];
        if (statement.kind == ts.SyntaxKind.VariableStatement){
          let name = statement.name.escapedText
          let exportNode:Graph.JsExportNode = {
            variableName: name,
            value: undefined,
            language:"javascript",
            code: node, 
            kind: 'export',
            antecedents:[node]
            };
          dependencies.push(exportNode);
          node.exportedVariables.push(exportNode.variableName);
        }
        walk(statement)
      }
```
And then we walk the other statements. If the statement is `(var a = 1;)` it drops out immediately because there are no children.